src/gsy_e/models/market/two_sided.py

```diff
@@ -440,13 +441,28 @@ def _replace_offers_bids_with_residual_in_recommendations_list(
         replaced with corresponding residual offer/bid
         """
 
+        def _adapt_matching_requirements_in_residuals(recommendation):
+            if "energy" in (recommendation.get("matching_requirements") or {}).get(
+                    "bid_requirement", {}):
+                for index, requirement in enumerate(recommendation["bid"]["requirements"]):
+                    if requirement == recommendation["matching_requirements"]["bid_requirement"]:
+                        bid_requirement = deepcopy(requirement)
+                        bid_requirement["energy"] -= bid_trade.traded_energy
+                        recommendation["bid"]["requirements"][index] = bid_requirement
```

Comment:
Just out of interest: The garbage collector will delete the "old" requirement , right?

Reply:
As far as I see, it won't be referenced anywhere so it should be deleted.